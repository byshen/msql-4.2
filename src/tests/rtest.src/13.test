#
# Test CIDR type handling

create table routes (prefix CIDR4, descr CHAR(20))\p\g

insert into routes values ('10.0.0.0/8', 'Net-10 aggregate') \p\g

insert into routes values ('10.1.0.0/16', 'Net-10 block 1') \p\g
insert into routes values ('10.2.0.0/16', 'Net-10 block 2') \p\g
insert into routes values ('10.3.0.0/16', 'Net-10 block 3') \p\g
insert into routes values ('10.4.0.0/16', 'Net-10 block 4') \p\g

insert into routes values ('10.1.1.0/24', 'Net-10 block 1.1') \p\g
insert into routes values ('10.1.2.0/24', 'Net-10 block 1.2') \p\g
insert into routes values ('10.1.3.0/24', 'Net-10 block 1.3') \p\g
insert into routes values ('10.1.4.0/24', 'Net-10 block 1.4') \p\g

insert into routes values ('10.2.1.0/24', 'Net-10 block 2.1') \p\g
insert into routes values ('10.2.2.0/24', 'Net-10 block 2.2') \p\g
insert into routes values ('10.2.3.0/24', 'Net-10 block 2.3') \p\g
insert into routes values ('10.2.4.0/24', 'Net-10 block 2.4') \p\g

insert into routes values ('10.1.1.1/32', 'Net-10 block 1.1.1') \p\g
insert into routes values ('10.1.1.2/32', 'Net-10 block 1.1.2') \p\g
insert into routes values ('10.1.1.3/32', 'Net-10 block 1.1.3') \p\g
insert into routes values ('10.1.1.4/32', 'Net-10 block 1.1.4') \p\g

insert into routes values ('10.1.2.1/32', 'Net-10 block 1.2.1') \p\g
insert into routes values ('10.1.2.2/32', 'Net-10 block 1.2.2') \p\g
insert into routes values ('10.1.2.3/32', 'Net-10 block 1.2.3') \p\g
insert into routes values ('10.1.2.4/32', 'Net-10 block 1.2.4') \p\g

select * from routes where prefix = '10.1.2.0/24' \p\g
select * from routes where prefix <= '10.1.2.3/32' order by prefix \p\g
select * from routes where prefix > '10.1.1.0/24' order by prefix \p\g
select * from routes where prefix < '10.1.1.1/24' order by prefix desc limit 1\p\g
