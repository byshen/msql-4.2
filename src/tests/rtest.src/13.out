
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     -> 
Query buffer
------------

create table routes (prefix CIDR4, descr CHAR(20))
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.0.0.0/8', 'Net-10 aggregate') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.1.0.0/16', 'Net-10 block 1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.2.0.0/16', 'Net-10 block 2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.3.0.0/16', 'Net-10 block 3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.4.0.0/16', 'Net-10 block 4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.1.1.0/24', 'Net-10 block 1.1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.2.0/24', 'Net-10 block 1.2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.3.0/24', 'Net-10 block 1.3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.4.0/24', 'Net-10 block 1.4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.2.1.0/24', 'Net-10 block 2.1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.2.2.0/24', 'Net-10 block 2.2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.2.3.0/24', 'Net-10 block 2.3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.2.4.0/24', 'Net-10 block 2.4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.1.1.1/32', 'Net-10 block 1.1.1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.1.2/32', 'Net-10 block 1.1.2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.1.3/32', 'Net-10 block 1.1.3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.1.4/32', 'Net-10 block 1.1.4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into routes values ('10.1.2.1/32', 'Net-10 block 1.2.1') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.2.2/32', 'Net-10 block 1.2.2') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.2.3/32', 'Net-10 block 1.2.3') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into routes values ('10.1.2.4/32', 'Net-10 block 1.2.4') 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
select * from routes where prefix = '10.1.2.0/24' 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +-----------------------+----------------------+
 | prefix                | descr                |
 +-----------------------+----------------------+
 | 10.1.2.0/24           | Net-10 block 1.2     |
 +-----------------------+----------------------+



mSQL > 
Query buffer
------------
select * from routes where prefix <= '10.1.2.3/32' order by prefix 
[continue]
    -> 
Query OK.  4 row(s) modified or retrieved.

 +-----------------------+----------------------+
 | prefix                | descr                |
 +-----------------------+----------------------+
 | 10.0.0.0/8            | Net-10 aggregate     |
 | 10.1.0.0/16           | Net-10 block 1       |
 | 10.1.2.0/24           | Net-10 block 1.2     |
 | 10.1.2.3/32           | Net-10 block 1.2.3   |
 +-----------------------+----------------------+



mSQL > 
Query buffer
------------
select * from routes where prefix > '10.1.1.0/24' order by prefix 
[continue]
    -> 
Query OK.  4 row(s) modified or retrieved.

 +-----------------------+----------------------+
 | prefix                | descr                |
 +-----------------------+----------------------+
 | 10.1.1.1/32           | Net-10 block 1.1.1   |
 | 10.1.1.2/32           | Net-10 block 1.1.2   |
 | 10.1.1.3/32           | Net-10 block 1.1.3   |
 | 10.1.1.4/32           | Net-10 block 1.1.4   |
 +-----------------------+----------------------+



mSQL > 
Query buffer
------------
select * from routes where prefix < '10.1.1.1/24' order by prefix desc limit 1
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +-----------------------+----------------------+
 | prefix                | descr                |
 +-----------------------+----------------------+
 | 10.1.0.0/16           | Net-10 block 1       |
 +-----------------------+----------------------+



mSQL > 
Bye!

